src/Pool.ts

```diff
@@ -514,10 +514,25 @@ export class Pool<T> {
     this._executeEventHandlers('createRequest', eventId);
 
     const pendingCreate = new PendingOperation<T>(this.createTimeoutMillis);
+
+    // Note if an error occurs (likely a create timeout) and remove this
+    // creation from list of pending creations so we try to create a new one.
+    let cancelled = false;
```

Comment:
Could pendingCreate.isRejected be used instead of separate cancelled flag here?

Reply:
Yes, thanks for pointing this out! I updated the commit to simplify it.